### DISCUSSION TITLE
How to persist a variable in the pipeline?

### DISCUSSION THREAD

**Discussion by jiekechoo** [edgexfoundry/community#257] [Q&A] [answered]
When the pipeline started, needs to persist a variable `key=value` for reuse purpose, just for memory usage. 
1. add value
```
ctx.AddValue("key","value")
ctx.ApplyValues("key")
```
2. next time get value
```
v,err := ctx.GetValue("key")
```
It returns error=false.

Is there any function or api can handler this?

**Comment by jiekechoo** [comment, marked as answer]
The `Context Storage` functions only support in the current pipeline, can't store global variables, so it is not reusable.

Finally, I use the `MetricsManager` function in main() and the pipeline, solved.

**Reply by lenny-goodell** [reply]
Also, the struct for your custom pipeline function is the place to persist value.

**Reply by jiekechoo** [reply]
So, where and when should I initiate the struct for the custom pipeline?
